test/e2e/pkg/testnet.go

```diff
@@ -100,17 +103,28 @@ type Node struct {
 // The testnet generation must be deterministic, since it is generated
 // separately by the runner and the test cases. For this reason, testnets use a
 // random seed to generate e.g. keys.
-func LoadTestnet(file string) (*Testnet, error) {
+func LoadTestnet(file string, ifd InfrastructureData) (*Testnet, error) {
 	manifest, err := LoadManifest(file)
 	if err != nil {
 		return nil, err
 	}
 	dir := strings.TrimSuffix(file, filepath.Ext(file))
 
 	// Set up resource generators. These must be deterministic.
-	netAddress := networkIPv4
-	if manifest.IPv6 {
-		netAddress = networkIPv6
+	var netAddress string
+	switch ifd.Provider {
+	case "docker":
+		netAddress = dockerIPv4CIDR
+		if manifest.IPv6 {
+			netAddress = dockerIPv6CIDR
+		}
+	default:
+		// TODO(williambanfield): add list of CIDR blocks to the infrastructure
+		// data struct to allow tighter validation of IP addresses.
+		netAddress = globalIPv4CIDR
+		if manifest.IPv6 {
+			netAddress = globalIPv6CIDR
```

Comment:
What is this used for (in the case of DO)?

Reply:
It's validation that all of the IPs in the generated network are within the CIDR. For docker, this is constrained but for DO at the moment, it's a 'slash 0' meaning it doesn't actually do anything for the DO case.

Reply:
I see, makes sense